### ISSUE TITLE
ed25519 crate

### ISSUE THREAD

**Issue by tarcieri** [RustCrypto/signatures#22]
Now that the `signature` crate is released in some form and being used, the next step I think makes sense is to work on the `ed25519` crate (which @newpavlov has already reserved).

I think the best way to keep it simple for an initial release would be to focus exclusively on an `ed25519::Signature` type which impls the [`signature::Signature`](https://docs.rs/signature/latest/signature/trait.Signature.html) trait and can be shared across crates that produce and consume Ed25519 signatures in conjunction with the [`signature::Signer`](https://docs.rs/signature/latest/signature/trait.Signer.html) and [`signature::Verifier`](https://docs.rs/signature/latest/signature/trait.Signer.html) traits.

A unified public key type would be nice as well, although the `signature` crate presently has no traits relating to public keys, so this seems like a good issue to punt on until the question of the relationship between the `signature` crate and public keys is addressed.

An `ed25519::Seed` type containing the derivation (via SHA-512) secret for the full keypair might be interesting as well, but also something that seems worthwhile to punt on in the short term.

**Comment by tarcieri** [comment]
Here's a quick sketch:

```rust
pub const SIZE: usize = 64;

#[derive(Clone, Debug)]
pub struct Signature(pub [u8; SIZE]);

// Note: external consumers of the crate will refer to these as `ed25519::Signature` and `signature::Signature`
impl signature::Signature for Signature {
    fn from_bytes(bytes: [u8; SIZE]) -> Self {
        Signature(bytes)
    }

    fn as_slice(&self) -> &[u8; SIZE] {
        &self.0
    }

    fn into_vec(self) -> [u8; SIZE] {
        Vec::from(self.0)
    }
}

impl AsRef<[u8]> for Signature {
    fn as_ref(&self) -> &[u8] {
        self.0.as_ref()
    }
}
```

**Comment by tarcieri** [comment]
@newpavlov well, there's a basic crate in place. Anything else you think we should do before publishing it?

**Comment by newpavlov** [comment]
I don't have any suggestions right now, so I think you can publish it.

**Comment by tarcieri** [comment]
@newpavlov can you add me as an owner?

**Comment by newpavlov** [comment]
Oh, I forgot about that. You now should have publishing rights!

**Comment by tarcieri** [comment]
@newpavlov thanks! Released as v0.1.0
